internal/cmd/precise-code-intel-test/internal/parallel.go

```diff
@@ -121,3 +111,43 @@ func RunParallel(maxConcurrency int, fns []FnPair) error {
 		return nil
 	})
 }
+
+type pendingMap struct {
+	m       sync.RWMutex
```

Comment:
ok, just a nit:
```suggestion
     sync.RWMutex
```

that way you don't need the `m.m.` stutter. you can just `m.Lock()`